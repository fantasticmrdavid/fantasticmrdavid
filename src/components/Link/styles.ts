import styled from 'styled-components';
import theme from 'styles/theme';
import { media } from 'styles/utils';

export const Link = styled.a`
  color: ${theme.colorWhite};
  text-decoration: none;
  border-bottom: ${`1px solid ${theme.colorWhite}`};
  transition: 0.3s;

  ${media.medium`
    &:hover {
      border-bottom-color: ${theme.colorRed};
    }
  `};
`;
