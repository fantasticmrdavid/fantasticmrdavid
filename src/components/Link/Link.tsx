import React, { memo } from 'react';
import { Link as StyledLink } from './link.styles';

interface LinkProps {
  children?: any,
  href: string,
  onClick?: (...args: any[]) => any,
  target?: string,
  rel?: string,
}

export const Link = memo((props: LinkProps) => <StyledLink {...props} />);
