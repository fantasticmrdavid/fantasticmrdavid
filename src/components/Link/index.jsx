import React from 'react';
import * as styles from './styles';

export default (props) => {
  const { Link } = styles;
  return <Link {...props} />;
};
