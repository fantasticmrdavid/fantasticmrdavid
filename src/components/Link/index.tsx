import React from 'react';
import * as styles from './styles';

interface Props {
  children?: any,
  href: string,
  onClick?: (...args: any[]) => any,
  target?: string,
  rel?: string,
}

export default (props: Props) => {
  const { Link } = styles;
  return <Link {...props} />;
};
