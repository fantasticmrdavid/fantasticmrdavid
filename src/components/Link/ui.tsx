import React, { memo } from 'react';
import { Link } from './styles';

interface Props {
  children?: any,
  href: string,
  onClick?: (...args: any[]) => any,
  target?: string,
  rel?: string,
}

export default memo((props: Props) => <Link {...props} />);
