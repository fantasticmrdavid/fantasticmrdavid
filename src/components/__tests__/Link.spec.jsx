import React from 'react';
import renderer from 'react-test-renderer';
import Link from 'components/Link';

describe('Component -> Link', () => {
  it('should render the Link without throwing an error', () => {
    expect(renderer.create(<Link href="/" />)).toMatchSnapshot();
  });
});
