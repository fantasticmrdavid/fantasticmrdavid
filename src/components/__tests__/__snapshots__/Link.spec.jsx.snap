// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> Link should render the Link without throwing an error 1`] = `
<a
  className="styles__Link-sc-1rd8ebj-0 fmVTeT"
  href="/"
/>
`;
